airflow/stats.py

```diff
@@ -250,43 +298,107 @@ def test(self, stat):
             return True  # default is all metrics allowed
 
 
+def prepare_stat_with_tags(fn: T) -> T:
+    """Add tags to stat with influxdb standard format if influxdb_tags_enabled is True."""
+
+    @wraps(fn)
+    def wrapper(self, stat=None, *args, tags=None, **kwargs):
+        if self.influxdb_tags_enabled:
+            if stat is not None and tags is not None:
+                for k, v in tags.items():
+                    if not set(",=").intersection(set(v + k)):
+                        stat += f",{k}={v}"
+                    else:
+                        log.error("Dropping invalid tag: %s=%s.", k, v)
+        return fn(self, stat, *args, tags=tags, **kwargs)
+
+    return cast(T, wrapper)
+
+
 class SafeStatsdLogger:
     """StatsD Logger."""
 
-    def __init__(self, statsd_client, allow_list_validator=AllowListValidator()):
+    def __init__(
+        self,
+        statsd_client,
+        allow_list_validator=AllowListValidator(),
+        aggregation_optimizer_enabled=False,
+        influxdb_tags_enabled=False,
+    ):
         self.statsd = statsd_client
         self.allow_list_validator = allow_list_validator
+        self.aggregation_optimizer_enabled = aggregation_optimizer_enabled
+        self.influxdb_tags_enabled = influxdb_tags_enabled
 
+    @prepare_stat_with_tags
     @validate_stat
-    def incr(self, stat, count=1, rate=1, tags: dict[str, str] | None = None):
+    def incr(
+        self,
+        stat,
+        count=1,
+        rate=1,
+        *,
+        tags: dict[str, str] | None = None,
+    ):
```

Comment:
Add types to other arguments while we’re changing these? (Same for other functions)